### PR TITLE
Allow mixed `str`/`dict` inputs/outputs to tasks

### PR DESCRIPTION
**Pull request by GeigerJ2**

### CODE REVIEW

File: aiida_workgraph/utils/__init__.py [hunk 1]
```diff
@@ -766,18 +766,22 @@ def validate_task_inout(inout_list: list[str | dict], list_type: str) -> list[di
     if the former convert them to a list of `dict`s with `name` as the key.
 
     :param inout_list: The input/output list to be validated.
-    :param list_type: "input" or "output" to indicate what is to be validated.
-    :raises TypeError: If a list of mixed or wrong types is provided to the task
+    :param list_type: "input" or "output" to indicate what is to be validated for better error message.
```
Comment (superstar54) [inline aiida_workgraph/utils/__init__.py]:
```suggestion
    :param list_type: "inputs" or "outputs" to indicate what is to be validated for better error message.
```

**Review by superstar54 — CHANGES_REQUESTED** [review]
@GeigerJ2 , thanks for the work!

Could you also update the docs to tell the user that we have these options.
e.g., https://aiida-workgraph.readthedocs.io/en/latest/concept/autogen/task.html

**Comment by GeigerJ2** [comment]
> @GeigerJ2 , thanks for the work!
> 
> Could you also update the docs to tell the user that we have these options. e.g., [aiida-workgraph.readthedocs.io/en/latest/concept/autogen/task.html](https://aiida-workgraph.readthedocs.io/en/latest/concept/autogen/task.html)

Cheers! Yes, good catch, I will add it!

**Review by superstar54 — APPROVED** [review]
LGTM!

**Pull request merged**